### PR TITLE
Feature: Allow `minLines` in `SelectableText`

### PR DESCRIPTION
**Pull request by Inconnu08**
## Description

Allow `SelectableText` to have `minLines` property.

## Related Issues

#45043

## Tests

I added the following tests:

- `minLines cannot be greater than maxLines`

- `Selectable height with minLine`

- Modifies `SelectableText implements debugFillProperties` test

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/selectable_text.dart [hunk 1]
```diff
@@ -582,6 +598,7 @@ class _SelectableTextState extends State<SelectableText> with AutomaticKeepAlive
         autofocus: widget.autofocus,
         forceLine: false,
         toolbarOptions: widget.toolbarOptions,
+        minLines: widget.minLines,
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/selectable_text.dart]: Not to be addressed here, but I just noticed that `DefaultTextStyle` does not have a `minLines` parameter :)

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 2]
```diff
@@ -177,12 +177,13 @@ void main() {
     debugResetSemanticsIdCounter();
   });
 
-  Widget selectableTextBuilder({String text = '', int maxLines = 1}) {
+  Widget selectableTextBuilder({String text = '', int maxLines = 1, int minLines}) {
```
Comment (shihaohong) [inline packages/flutter/test/widgets/selectable_text_test.dart]: nit: break this out
```suggestion
  Widget selectableTextBuilder({
    String text = '',
    int maxLines = 1,
    int minLines,
  }) {
```

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 3]
```diff
@@ -177,12 +177,13 @@ void main() {
...
     return boilerplate(
       child: SelectableText(
         text,
         style: const TextStyle(color: Colors.black, fontSize: 34.0),
         maxLines: maxLines,
+        minLines: minLines
```
Comment (shihaohong) [inline packages/flutter/test/widgets/selectable_text_test.dart]: nit: trailing comma
```suggestion
        minLines: minLines,
```

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 4]
```diff
@@ -1155,6 +1156,39 @@ void main() {
...
+              'abcd',
+              minLines: 4,
+              maxLines: 3,
+            ),
+          ),
+        ),
+      );
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains("minLines can't be greater than maxLines"));
+    }
```
Comment (shihaohong) [inline packages/flutter/test/widgets/selectable_text_test.dart]: Add a line after this one to make sure the test fails if the assert is removed:
```dart
    fail('An assert should be triggered when minLines is greater than maxLines');
```

File: packages/flutter/test/widgets/selectable_text_test.dart [hunk 5]
```diff
@@ -1155,6 +1156,39 @@ void main() {
...
+            child: SelectableText(
+              'abcd',
+              minLines: 4,
+              maxLines: 3,
+            ),
+          ),
+        ),
+      );
+    } on AssertionError catch (e) {
+      expect(e.toString(), contains("minLines can't be greater than maxLines"));
```
Comment (shihaohong) [inline packages/flutter/test/widgets/selectable_text_test.dart]: Add a `return;` after this line to make sure the failure is not hit if the `AssertionError` is correctly triggered.

**Review by shihaohong — CHANGES_REQUESTED** [review]
Some nits and a request to improve one of the tests. Thanks for creating a new PR for this feature!

File: packages/flutter/test/widgets/selectable_text_test.dart (hunk 4 above)
Comment (shihaohong) [reply packages/flutter/test/widgets/selectable_text_test.dart]: You can check to see if it properly fails by removing the assert from your code and seeing if the test case fails

File: packages/flutter/lib/src/material/selectable_text.dart (hunk 1 above)
Comment (Inconnu08) [reply packages/flutter/lib/src/material/selectable_text.dart]: Yes, is that a different issue/feature?

Comment (Inconnu08) [reply packages/flutter/lib/src/material/selectable_text.dart]: Yeah, you should probably open a new issue for that.

**Review by shihaohong — APPROVED** [review]
LGTM

**Review by shihaohong — COMMENTED** [review]
I just realized, we forgot to modify the `SelectableText implements debugFillProperties` test. Could you go ahead and do that?

**Comment by Inconnu08** [comment]
> I just realized, we forgot to modify the `SelectableText implements debugFillProperties` test. Could you go ahead and do that?

Oops, yes give me a minute.

**Review by shihaohong — APPROVED** [review]

**Pull request merged**